src/openrct2/world/Climate.cpp

```diff
@@ -275,14 +284,9 @@ FilterPaletteID ClimateGetWeatherGloomPaletteId(const WeatherState& state)
     return paletteId;
 }
 
-uint32_t ClimateGetWeatherSpriteId(const WeatherState& state)
+uint32_t ClimateGetWeatherSpriteId(const WeatherType weatherType)
 {
-    uint32_t spriteId = SPR_WEATHER_SUN;
-    if (EnumValue(state.weatherType) < std::size(kClimateWeatherTraits))
-    {
-        spriteId = kClimateWeatherTraits[EnumValue(state.weatherType)].spriteId;
-    }
-    return spriteId;
+    return kClimateWeatherTraits[EnumValue(weatherType)].spriteId;
```

Comment:
This does assume this variable will never be out of bounds. Can we safely assume that?

Reply:
I'd like to, yes. If we can't trust the WeatherType enum to be enforced, that's a type problem, imo.

Reply:
If you like, we could turn `kClimateWeatherTraits` into a `std::array` so we keep it bounds-checked?